src/Lucene.Net/Util/AttributeSource.cs

```diff
@@ -51,8 +51,8 @@ public abstract class AttributeFactory
 			private sealed class DefaultAttributeFactory:AttributeFactory
 			{
                 // This should be WeakDictionary<T, WeakReference<TImpl>> where typeof(T) is Attribute and TImpl is typeof(AttributeImpl)
-			    private static readonly WeakDictionary<Type, WeakReference> attClassImplMap =
-			        new WeakDictionary<Type, WeakReference>();
+			    private static readonly ConditionalWeakTable<Type, WeakReference> attClassImplMap =
```

Comment:
This can be a normal dictionary, the `Type` is never going away

Reply:
changed to `ConcurrentDictionary`